_posts/2023-06-15-user-survey-2023.md

```diff
@@ -0,0 +1,240 @@
+
+* support for the Vivado simulator (xsim) [which unfortunately does not implement the VPI or VHPI interface], and
+* Better support for GHDL [which is partially due to the fact that GHDL is uniquely using the VPI interface for VHDL, as well as missing features in GHDL itself].
+
+## Verification IP or cocotb extensions
+
+**Most cocotb users use cocotb-test and a coverage extension, such as pyvsc or cocotb-coverage. Python implementations of UVM are popular with cocotb users as well, as is (AXI) bus IP.**
+
+Cocotb is intentionally not solving every possible problem: it's extensible and encourages users to choose their preferred extensions from the Python package ecosystem.
+
+For users who need more than cocotb's Makefile-based build system, cocotb-test is the go-to choice.
+Functional coverage and constraint-random test generation are next on the list, with users mostly relying on pyvsc or cocotb-coverage.
+
+As we'll see below, the availability of verification IP (VIP) is one of the weak spots of cocotb.
+Still, high-quality VIP is available, and many users resort to cocotbext-axi for AXI bus adapters or cocotbext-spi for SPI adapters.
+Additionally, cocotb-bus provides a further set of bus interfaces which is commonly used.
+
+Finally, despite many users choosing cocotb because it's *not* UVM, a significant percentage of the respondents choose a UVM implementation in Python, with pyuvm being more popular than uvm-python, followed by in-house UVM-like methodologies.
+
+## Are users happy with cocotb?
+
+**Users love cocotb!**
+
+Cocotb users overwhelmingly love using it, and would recommend it to others.
+That's enouraging to hear, and indicates that we're on the right path.
+We are still not at the point where every cocotb user is fully satisfied (to the extend that's ever possible), but overall the answers are very encouraging.
+
+<div class="row p-5">
+  <div class="col">
+    <img src="/assets/blog/2023-06-15-user-survey-2023/12-enjoy.svg" class="img-fluid mx-auto d-block" alt="How much do you enjoy using cocotb (5 == most, 1 == not at all)">
+  </div>
+  <div class="col">
+    <img src="/assets/blog/2023-06-15-user-survey-2023/13-recommend.svg" class="img-fluid mx-auto d-block" alt="How likely are you to recommend cocotb to others?">
+  </div>
+</div>
+
+Looking at the more detailed answers, we see common themes emerge:
+
+Users love cocotb because
+* It is Python
+* It is open source
+* It integrates into the Python ecosystem
+
+Users are least happy about
+* The (lack of) tutorials and learning resources
+* The (lack of) documentation
+* The Makefile-based testrunner
+* The availability of verification IP
+
+Interstingly, roughly an equal amount of users love and hate the cocotb installation through `pip install cocotb`.
+We'd love to  hear more from those users who struggle with the process, since we invested significantly into this area in recent releases.
```

Comment:
```suggestion
We'd love to hear more from those users who struggle with the process, since we invested significantly into this area in recent releases.
```